src/charm.py

```diff
@@ -359,6 +364,16 @@ def join_unit_to_cluster(self) -> None:
                 # Stop GR for cases where the instance was previously part of the cluster
                 # harmless otherwise
                 self._mysql.stop_group_replication()
+
+                # If instance already in cluster, rescan cluster on primary before
+                # attempting to add the instance to the cluster
+                cluster_status = self._mysql.get_cluster_status(from_instance=cluster_primary)
+                instances = cluster_status["defaultreplicaset"]["topology"].keys()
+                if instance_label in instances:
+                    self._mysql.rescan_cluster(
```

Comment:
I wonder if the rescan is doing some harm here, since can remove not only this instance but others. Maybe use `remove_instance` api method instead?

Reply:
addressed in [113db7d](https://github.com/canonical/mysql-k8s-operator/pull/433/commits/113db7d74b195d2a4b03c69ec185b6c3614430ae)